core/chain_util_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type diffTest struct {
+	ParentTimestamp    uint64
+	ParentDifficulty   *big.Int
+	CurrentTimestamp   uint64
+	CurrentBlocknumber *big.Int
+	CurrentDifficulty  *big.Int
+}
+
+func (d *diffTest) UnmarshalJSON(b []byte) (err error) {
+	var ext struct {
+		ParentTimestamp    string
+		ParentDifficulty   string
+		CurrentTimestamp   string
+		CurrentBlocknumber string
+		CurrentDifficulty  string
+	}
+	if err := json.Unmarshal(b, &ext); err != nil {
+		return err
+	}
+
+	d.ParentTimestamp = common.String2Big(ext.ParentTimestamp).Uint64()
+	d.ParentDifficulty = common.String2Big(ext.ParentDifficulty)
+	d.CurrentTimestamp = common.String2Big(ext.CurrentTimestamp).Uint64()
+	d.CurrentBlocknumber = common.String2Big(ext.CurrentBlocknumber)
+	d.CurrentDifficulty = common.String2Big(ext.CurrentDifficulty)
+
+	return nil
+}
+
+func loadDiffTest() (map[string]diffTest, error) {
+	file, err := os.Open("../tests/files/BasicTests/difficulty.json")
+	if err != nil {
+		return nil, err
+	}
+	content, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+
+	tests := make(map[string]diffTest)
+	if err := json.Unmarshal(content, &tests); err != nil {
+		return nil, err
+	}
+
+	return tests, nil
+}
```

Comment:
`json.NewDecoder(file).Decode(&tests)` saves you the trouble of doing `ReadAll`.
If this were in package tests, you could even to `readJsonFile("files/...", &tests)` and it would show line numbers in decode errors.